src/models/residence.js

```diff
@@ -0,0 +1,140 @@
+import { relationshipOptions, roleOptions } from 'constants/enums/residenceOptions'
+
+import address from 'models/shared/locations/address'
+import name from 'models/shared/name'
+import phone from 'models/shared/phone'
+import email from 'models/shared/email'
+
+import { today, dateWithinRange } from 'helpers/date'
+
+const residenceRequiresReference = (dates = {}) => {
+  const { from, present } = dates
+  const to = present ? today.toObject() : dates.to
+  const referenceTimeframe = { years: 3 }
+
+  return dateWithinRange(to, referenceTimeframe)
+    || dateWithinRange(from, referenceTimeframe)
+}
+
+const residence = {
+  Dates: {
+    presence: true,
+    daterange: true,
+  },
+  Address: {
+    presence: true,
+    location: { validator: address },
+  },
+
+  AlternateAddress: {
+    // TODO - currently no validation in place
```

Comment:
Yeah, definitely want USPS validation on all the various alternate addresses, along with all the usual validation.